### PR TITLE
Scripts inventory plugin, add examples

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

**Review by tremble — APPROVED** [review]

File: lib/ansible/plugins/inventory/script.py [hunk 1]
```diff
@@ -23,14 +23,135 @@
...
         - The plugin does not cache results because external inventory scripts are responsible for their own caching.
         - To write your own inventory script see (R(Developing dynamic inventory,developing_inventory) from the documentation site.
         - To find the scripts that used to be part of the code release, go to U(https://github.com/ansible-community/contrib-scripts/).
 '''
 
+EXAMPLES = r'''
+
+### simple bash script
+
+   #! /bin/bash
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
   #!/usr/bin/env bash
```

File: lib/ansible/plugins/inventory/script.py [hunk 2]
```diff
@@ -23,14 +23,135 @@
...
+     fi
+   else
+     echo "Invalid option: use --list or --host <hostname>"
+     exit 1
+   fi
+
+
+### python example with ini config
+
+    !/usr/bin/env python
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
    #!/usr/bin/env python
```

File: lib/ansible/plugins/inventory/script.py [hunk 3]
```diff
@@ -23,14 +23,135 @@
...
+    """
+    # ansible_inventory.py
+    """
+    import argparse
+    import json
+    import os.path
+    import sys
+
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
from configparser import ConfigParser
```

File: lib/ansible/plugins/inventory/script.py [hunk 4]
```diff
@@ -23,14 +23,135 @@
...
+    def get_api_data(namespace: str, pretty=False) -> str:
+        """
+        :param namespace: parameter for our custom api
+        :param pretty: Human redable JSON vs machine readable
+        :return: JSON string
+        """
+        found_data = list(MyInventoryAPI(namespace))
+        hostvars = {}
+        data = { '_meta': { 'hostvars': {}},
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
        data = { '_meta': { 'hostvars': {}},}
```

File: lib/ansible/plugins/inventory/script.py [hunk 5]
```diff
@@ -23,14 +23,135 @@
...
+            data[group]['children'] = found_data[group].get('subgroups', [])
+
+        for host_data in found_data['hosts']:
+            for name in host_data.items():
+                # turn info into vars
+                data['_meta'][name] = found_data[name].get('info', {})
+                # set ansible_host if possible
+                if 'address' in found_data[name]:
+                    data[name]['_meta']['ansible_host'] = found_data[name]['address']
+        }
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion

```

File: lib/ansible/plugins/inventory/script.py [hunk 6]
```diff
@@ -23,14 +23,135 @@
...
+        arg_parser = argparse.ArgumentParser( description=__doc__, prog=__file__)
+        arg_parser.add_argument('--pretty', action='store_true', default=False, help="Pretty JSON")
+        mandatory_options = arg_parser.add_mutually_exclusive_group()
+        mandatory_options.add_argument('--list', action='store', nargs="*", help="Get inventory JSON from our API")
+        mandatory_options.add_argument('--host', action='store',
+                                       help="Get variables for specific host, not used but kept for compatability")
+
+        try:
+            config = load_config()
+            namespace = config.get('defaults', 'namespace')
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
            namespace = config.get('DEFAULT', 'namespace')
```

File: lib/ansible/plugins/inventory/script.py [hunk 7]
```diff
@@ -23,14 +23,135 @@
...
+
+            args = arg_parser.parse_args()
+            if args.host:
+                print('{"_meta":{}}')
+                sys.stderr.write('This script already provides _meta via --list, so this option is really ignored')
+            elif len(args.list) >= 0:
+                print(get_api_data(namespace, args.pretty))
+            else:
+                raise ValueError("Valid options are -- list or --host <HOSTNAME>")
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
                raise ValueError("Valid options are --list or --host <HOSTNAME>")
```

File: lib/ansible/plugins/inventory/script.py [hunk 8]
```diff
@@ -23,14 +23,135 @@
...
+
+   #! /bin/bash
+
+   if [ "$1" == "--list" ]; then
+   cat<<EOF
+   {
+     "bash_hosts": {
+       "hosts": [
+         "myhost.domain.com",
+         "myhost2,domain.com"
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
         "myhost2.domain.com"
```

File: lib/ansible/plugins/inventory/script.py [hunk 9]
```diff
@@ -23,14 +23,134 @@
...
+
+    #!/usr/bin/env python
+    """
+    # ansible_inventory.py
+    """
+    import argparse
+    import json
+    import os.path
+    import sys
+from configparser import ConfigParser
```
Comment (Akasurde) [inline lib/ansible/plugins/inventory/script.py]:
```suggestion
    from configparser import ConfigParser
```

**Review by Akasurde — APPROVED** [review]

File: test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py [hunk 10]
```diff
@@ -164,7 +164,6 @@ def get_module_docs(self, path, contents):  # type: (str, str) -> t.Dict[str, t.
         """Return the module documentation for the given module contents."""
         module_doc_types = [
             'DOCUMENTATION',
-            'EXAMPLES',
```
Comment (felixfontein) [inline test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py]: This is a major change IMO and should not happen without a changelog fragment and sneaked into a docs PR.

**Review by felixfontein — CHANGES_REQUESTED** [review]

File: test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py (hunk 10 above)
Comment (felixfontein) [reply test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py]: (Also I think this change should not happen. The mechanism described in https://github.com/ansible/ansible/pull/82355#issuecomment-1841469933 should be used instead.)

Comment (bcoca) [reply test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py]: agreed, i was unaware of such

File: lib/ansible/plugins/inventory/script.py [hunk 11]
```diff
@@ -23,14 +23,134 @@
...
-          This is a performance optimization as the script would be called per host otherwise.
+          This is a performance optimization as the script would be called one additional time per host otherwise.
     notes:
         - Enabled in configuration by default.
         - The plugin does not cache results because external inventory scripts are responsible for their own caching.
         - To write your own inventory script see (R(Developing dynamic inventory,developing_inventory) from the documentation site.
         - To find the scripts that used to be part of the code release, go to U(https://github.com/ansible-community/contrib-scripts/).
 '''
 
+EXAMPLES = r'''#fmt: code
```
Comment (felixfontein) [inline lib/ansible/plugins/inventory/script.py]: The space after `#` is important (https://github.com/ansible/ansible/blob/6ebefaceb6cd0d4961776a94d63a71fc1fc28bc0/test/lib/ansible_test/_util/controller/sanity/yamllint/yamllinter.py#L173):
```suggestion
EXAMPLES = r'''# fmt: code
```

**Review by felixfontein — APPROVED** [review]

**Pull request merged**